File: backend/api/models/simulation.py
from pydantic import BaseModel
from typing import Literal

from api.models.block import ModelBlockConfig, SimulationBlockConfig, PlotBlockConfig


class SimulationRequest(BaseModel):
    diagram_id: str
    model_block: ModelBlockConfig
    simulation_block: SimulationBlockConfig
    plot_blocks: list[PlotBlockConfig] = []


class TimeSeriesPoint(BaseModel):
    t: float
    values: dict[str, float]


class SimulationResult(BaseModel):
    run_id: str
    status: Literal["running", "completed", "failed"]
    series: list[TimeSeriesPoint] = []
    error: str | None = None
    elapsed_seconds: float | None = None

File: frontend/src/types/simulation.ts
export interface TimeSeriesPoint {
  t: number;
  values: Record<string, number>;
}

export type RunStatus = 'idle' | 'queued' | 'running' | 'completed' | 'failed';

export interface SimRun {
  status: RunStatus;
  runId: string | null;
  series: TimeSeriesPoint[];
  error: string | null;
  elapsedSeconds: number | null;
}
